src/python/grpcio/grpc/beta/implementations.py

```diff
@@ -105,10 +144,35 @@ def metadata_call_credentials(metadata_plugin, name=None):
     A CallCredentials object for use in a GRPCCallOptions object.
   """
   if name is None:
-    name = metadata_plugin.__name__
+    name = metadata_plugin.__class__.__name__
```

Comment:
Instances don't have a __name__ in python 
http://stackoverflow.com/questions/14514838/why-do-python-instances-have-no-name-attribute
If the name option is None, this causes an error.